From cbc74d3a65068a5f54b663cc8fc6e15a4b5d081b Mon Sep 17 00:00:00 2001
From: Santhosh-Kumar-infosys <sakumar.ext@libertyglobal.com>
Date: Fri, 6 Feb 2026 15:57:19 +0530
Subject: ARRISEOS-48107 : Remove API Implementation

---
 MediaSession.cpp | 35 +++++++++++++++++++++++++++++++++--
 1 file changed, 33 insertions(+), 2 deletions(-)

diff --git a/MediaSession.cpp b/MediaSession.cpp
index 118e921..79716f2 100644
--- a/MediaSession.cpp
+++ b/MediaSession.cpp
@@ -699,7 +699,35 @@ ErrorExit:
 
 CDMi_RESULT MediaKeySession::Remove(void)
 {
-    return CDMi_S_FALSE;
+    m_eKeyState = KEY_CLOSED;
+
+    {
+        SafeCriticalSection systemLock(drmAppContextMutex_);
+        CleanLicenseStore();
+        CleanDecryptContexts();
+    }
+
+    if (pNexusMemory) {
+        NEXUS_Memory_Free(pNexusMemory);
+        pNexusMemory = nullptr;
+        mNexusMemorySize = 0;
+    }
+
+    if (mDmaJob) {
+        NEXUS_DmaJob_Destroy(mDmaJob);
+        mDmaJob = nullptr;
+    }
+
+    if (mDmaHandle) {
+        NEXUS_Dma_Close(mDmaHandle);
+        mDmaHandle = nullptr;
+    }
+
+    m_piCallback = nullptr;
+    m_fCommit = FALSE;
+    m_decryptInited = false;
+
+    return CDMi_SUCCESS;
 }
 
 CDMi_RESULT MediaKeySession::Close(void)
@@ -1064,7 +1092,8 @@ void MediaKeySession::CleanLicenseStore()
         } else {
             m_hasInMemoryLicenses = false;
         }
-    } else if ((m_licenseType == Temporary) && m_hasPersistentLicenses && (!m_kids.empty())) {
+    }
+	if (m_hasPersistentLicenses && (!m_kids.empty())) {
         DRM_DWORD removedCount{};
         DRM_DWORD licensesRemoved{};
         m_hasPersistentLicenses = false;
@@ -1091,6 +1120,8 @@ void MediaKeySession::CleanLicenseStore()
             }
         }
         LOGGER(LINFO_, "Number of licenses removed :%d",licensesRemoved);
+    } else {
+        LOGGER(LINFO_, "Persistent licenses not found");
     }
 }
 
-- 
2.43.0

