From a90050c620dfd6ea5c41178c2025ae9e4794f42e Mon Sep 17 00:00:00 2001
From: Mikolaj Staworzynski <mikolaj.staworzynski@redembedded.com>
Date: Thu, 14 Apr 2022 11:46:18 +0200
Subject: [PATCH] [ONEM-25037] enable stub generation for Register/Unregister

---
 interfaces/IContentDecryption.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/interfaces/IContentDecryption.h b/interfaces/IContentDecryption.h
index 48b3f4b..1d051ce 100644
--- a/interfaces/IContentDecryption.h
+++ b/interfaces/IContentDecryption.h
@@ -53,8 +53,8 @@ namespace Exchange {
                                               const Status status) = 0;
         };
 
-        virtual uint32_t Register(IContentDecryption::INotification* notification VARIABLE_IS_NOT_USED) { return Core::ERROR_NOT_SUPPORTED; };
-        virtual uint32_t Unregister(IContentDecryption::INotification* notification VARIABLE_IS_NOT_USED) { return Core::ERROR_NOT_SUPPORTED; };
+        virtual uint32_t Register(IContentDecryption::INotification* notification) = 0;
+        virtual uint32_t Unregister(IContentDecryption::INotification* notification) = 0;
     };
 
     class DataExchange : public Core::SharedBuffer {
-- 
2.25.1

