From c111420268353160ebb3e333e135359c7dae5b8a Mon Sep 17 00:00:00 2001
From: Dawid Trendota <dawid.trendota@consult.red>
Date: Thu, 9 Mar 2023 16:53:27 +0100
Subject: [PATCH] [OMWAPPI-1316] Implementation of IMediaSystemMetrics
 interface

---
 MediaSystem.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/MediaSystem.cpp b/MediaSystem.cpp
index 9da7f42..92e1f66 100644
--- a/MediaSystem.cpp
+++ b/MediaSystem.cpp
@@ -142,7 +142,7 @@ namespace CDMi {
 
 static const char *DRM_DEFAULT_REVOCATION_LIST_FILE="/tmp/revpackage.xml";
 
-class PlayReady : public IMediaKeys, public IMediaKeysExt {
+class PlayReady : public IMediaKeys, public IMediaKeysExt, public IMediaSystemMetrics {
 private:
     PlayReady (const PlayReady&) = delete;
     PlayReady& operator= (const PlayReady&) = delete;
@@ -1027,6 +1027,11 @@ public:
         return CDMi_SUCCESS;
     }
 
+    CDMi_RESULT GetMetrics (uint32_t bufferLength, uint8_t buffer[], uint32_t &count) {
+        count = 0;
+        return CDMi_METHOD_NOT_IMPLEMENTED;
+    }
+
 private:
     DRM_WCHAR* drmdir_;
 
-- 
2.25.1

