From df38d380430d32a829092587c4f0ab931f21642e Mon Sep 17 00:00:00 2001
From: Mikolaj Staworzynski <mikolaj.staworzynski@redembedded.com>
Date: Wed, 22 Mar 2023 19:03:14 +0100
Subject: [PATCH] [OMWAPPI-1314] Extend ThunderInterfaces ocdm API for metrics
 support

---
 interfaces/IOCDM.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/interfaces/IOCDM.h b/interfaces/IOCDM.h
index 0d1ccd7..b247884 100644
--- a/interfaces/IOCDM.h
+++ b/interfaces/IOCDM.h
@@ -34,11 +34,13 @@ enum OCDM_RESULT : uint32_t {
     OCDM_INVALID_SESSION = 0x80000003,
     OCDM_INVALID_DECRYPT_BUFFER = 0x80000004,
     OCDM_OUT_OF_MEMORY = 0x80000005,
+    OCDM_METHOD_NOT_IMPLEMENTED = 0x80000006,
     OCDM_FAIL = 0x80004005,
     OCDM_INVALID_ARG = 0x80070057,
     OCDM_SERVER_INTERNAL_ERROR = 0x8004C600,
     OCDM_SERVER_INVALID_MESSAGE = 0x8004C601,
     OCDM_SERVER_SERVICE_SPECIFIC = 0x8004C604,
+    OCDM_BUFFER_TOO_SMALL = 0x8004C605,
 };
 
 // ISession defines the interface towards a DRM context that can decrypt data
@@ -242,6 +244,12 @@ struct IAccessorOCDM : virtual public Core::IUnknown {
         uint8_t secureStoreHash[] /* @out @length:secureStoreHashLength */,
         uint16_t secureStoreHashLength)
         = 0;
+
+    virtual OCDM_RESULT GetSystemMetrics(
+        const std::string& keySystem,
+        uint8_t metrics[] /* @out @length:metricsLen */,
+        uint32_t metricsLen /* @in */,
+        uint32_t &count /* @out */) = 0;
 };
 
 class EXTERNAL KeyId {
-- 
2.25.1


