diff --git a/Source/core/Trace.h b/Source/core/Trace.h
index 65e8f69e..ec2e9911 100644
--- a/Source/core/Trace.h
+++ b/Source/core/Trace.h
@@ -43,6 +43,8 @@ namespace WPEFramework {
     }
 }

+#define TRACE_LOG_FLAG  "/tmp/wpeframeworkTraceEnable"
+
 #ifdef __WINDOWS__
 #define TRACE_PROCESS_ID ::GetCurrentProcessId()
 #define TRACE_THREAD_ID ::GetCurrentThreadId()
@@ -66,8 +68,10 @@ namespace WPEFramework {
 #else
 #define TRACE_FORMATTING_IMPL(fmt, ...)                                                                                                     \
     do {                                                                                                                                    \
-        ::fprintf(stderr, "\033[1;32m[%s:%d](%s)<PID:%d><TID:%d>" fmt "\n\033[0m", &__FILE__[WPEFramework::Core::FileNameOffset(__FILE__)], __LINE__, __FUNCTION__, TRACE_PROCESS_ID, TRACE_THREAD_ID, ##__VA_ARGS__);  \
-        fflush(stderr);                                                                                                                 \
+           if(0 == access(TRACE_LOG_FLAG, F_OK)) {            \
+               ::fprintf(stderr, "\033[1;32m[%s:%d](%s)<PID:%d><TID:%d>" fmt "\n\033[0m", &__FILE__[WPEFramework::Core::FileNameOffset(__FILE__)], __LINE__, __FUNCTION__, TRACE_PROCESS_ID, TRACE_THREAD_ID, ##__VA_ARGS__);  \
+        fflush(stderr); \
+       } \
     } while (0)
 #endif

