diff --git a/Monitor/Monitor.h b/Monitor/Monitor.h
index 740d059f..80043daa 100644
--- a/Monitor/Monitor.h
+++ b/Monitor/Monitor.h
@@ -816,10 +816,18 @@ namespace Plugin {
                 _adminLock.Unlock();
             }
 
+            void Activation (const string& callsign, PluginHost::IShell* service) override
+            {
+                StateChange(service);
+            }
             void Activated (const string& callsign, PluginHost::IShell* service) override
             {
                 StateChange(service);
             }
+            void Deactivation (const string& callsign, PluginHost::IShell* service) override
+            {
+                StateChange(service);
+            }
             void Deactivated (const string& callsign, PluginHost::IShell* service) override
             {
                 StateChange(service);
diff --git a/RDKShell/RDKShell.cpp b/RDKShell/RDKShell.cpp
index c1a5c22c..5592f3c9 100755
--- a/RDKShell/RDKShell.cpp
+++ b/RDKShell/RDKShell.cpp
@@ -981,10 +979,18 @@ namespace WPEFramework {
                 }
             }
         }
+        void RDKShell::MonitorClients::Activation(const string& callsign, PluginHost::IShell* service)
+        {
+            StateChange(service);
+        }
         void RDKShell::MonitorClients::Activated(const string& callsign, PluginHost::IShell* service)
         {
             StateChange(service);
         }
+        void RDKShell::MonitorClients::Deactivation(const string& callsign, PluginHost::IShell* service)
+        {
+            StateChange(service);
+        }
         void RDKShell::MonitorClients::Deactivated(const string& callsign, PluginHost::IShell* service)
         {
             StateChange(service);
diff --git a/RDKShell/RDKShell.h b/RDKShell/RDKShell.h
index 767bce3f..6fc8e2fb 100755
--- a/RDKShell/RDKShell.h
+++ b/RDKShell/RDKShell.h
@@ -399,7 +399,9 @@ namespace WPEFramework {
 
               private:
                   virtual void StateChange(PluginHost::IShell* shell);
+                  virtual void  Activation(const string& callSign,  PluginHost::IShell* plugin);
                   virtual void  Activated(const string& callSign,  PluginHost::IShell* plugin);
+                  virtual void  Deactivation(const string& callSign,  PluginHost::IShell* plugin);
                   virtual void  Deactivated(const string& callSign,  PluginHost::IShell* plugin);
                   virtual void  Unavailable(const string& callSign,  PluginHost::IShell* plugin);
 

