From 76d1743da7c63b72707c337a5f1507b068a1e515 Mon Sep 17 00:00:00 2001
From: tomasz-karczewski-red <tomasz.karczewski@redembedded.com>
Date: Mon, 15 May 2023 12:42:30 +0000
Subject: [PATCH] remove impl

---
 Source/cryptography/CMakeLists.txt | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Source/cryptography/CMakeLists.txt b/Source/cryptography/CMakeLists.txt
index 8b8e6b1..303ef8d 100644
--- a/Source/cryptography/CMakeLists.txt
+++ b/Source/cryptography/CMakeLists.txt
@@ -48,7 +48,7 @@ ProxyStubGenerator(
     OUTDIR "${CMAKE_CURRENT_BINARY_DIR}/generated/proxystubs"
 )
 
-add_subdirectory(implementation)
+#add_subdirectory(implementation)
 
 add_library(${TARGET} SHARED
     Module.cpp
@@ -70,7 +70,6 @@ target_link_libraries(${TARGET}
 target_include_directories(${TARGET}
     PUBLIC
         $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}>
-        $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/implementation>
         $<INSTALL_INTERFACE:include/${NAMESPACE}/cryptography>
 )
 
-- 
2.30.0

