Piotr Andrzejewski <piotr.andrzejewski@redembedded.com>
Wed, 16 Apr 2025 19:13:28 +0000

diff --git a/interfaces/IDRM.h b/interfaces/IDRM.h
index f125e9b..297d82f 100644
--- a/interfaces/IDRM.h
+++ b/interfaces/IDRM.h
@@ -608,6 +608,8 @@ public:
         = 0;
 
     virtual CDMi_RESULT CleanSecureStore(const std::string& keySystem) = 0;
+    virtual CDMi_RESULT InitializeCtx(const std::string& keySystem) = 0;
+    virtual CDMi_RESULT DeinitializeCtx(const std::string& keySystem, bool cleanOnDestroy) = 0;
 };
 
 struct IMediaSystemMetrics {
diff --git a/interfaces/IOCDM.h b/interfaces/IOCDM.h
index fce81f9..7e69256 100644
--- a/interfaces/IOCDM.h
+++ b/interfaces/IOCDM.h
@@ -263,6 +263,8 @@ struct IAccessorOCDM : virtual public Core::IUnknown {
         = 0;
 
     virtual OCDM_RESULT CleanSecureStore(const std::string& keySystem) = 0;
+    virtual OCDM_RESULT InitializeCtx(const std::string& keySystem) = 0;
+    virtual OCDM_RESULT DeinitializeCtx(const std::string& keySystem, bool cleanOnDestroy) = 0;
 };
 
 class EXTERNAL KeyId {
