From 32759f0b49ba3baf62c9a8f7adbbda10f8102256 Mon Sep 17 00:00:00 2001
From: aiswarya-krishnan-infosys <aikrishnan.ext@libertyglobal.com>
Date: Thu, 19 Jun 2025 10:03:58 +0530
Subject: [PATCH] [ONEM-41322] Retrieve license URL alongside challenge data

---
 interfaces/IDRM.h  | 2 +-
 interfaces/IOCDM.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/interfaces/IDRM.h b/interfaces/IDRM.h
index 8c4f1b7..62a6394 100644
--- a/interfaces/IDRM.h
+++ b/interfaces/IDRM.h
@@ -515,7 +515,7 @@ public:
 
     virtual CDMi_RESULT SetDrmHeader(const uint8_t drmHeader[], uint32_t drmHeaderLength) = 0;
 
-    virtual CDMi_RESULT GetChallengeDataExt(uint8_t* challenge, uint32_t& challengeSize, uint32_t isLDL) = 0;
+    virtual CDMi_RESULT GetChallengeDataExt(uint8_t* challenge, uint32_t& challengeSize, uint32_t isLDL, uint8_t* url, uint32_t& urlsize) = 0;
 
     virtual CDMi_RESULT CancelChallengeDataExt() = 0;
 
diff --git a/interfaces/IOCDM.h b/interfaces/IOCDM.h
index 7fa6c14..803ec23 100644
--- a/interfaces/IOCDM.h
+++ b/interfaces/IOCDM.h
@@ -168,7 +168,7 @@ struct ISessionExt : virtual public Core::IUnknown {
 
     virtual OCDM_RESULT GetChallengeDataExt(uint8_t* challenge /* @inout @length:challengeSize */,
         uint16_t& challengeSize /* @inout */,
-        uint32_t isLDL)
+        uint32_t isLDL, uint8_t* url /* @out @length:urlSize */, uint16_t& urlSize /* @inout */)
         = 0;
 
     virtual OCDM_RESULT CancelChallengeDataExt() = 0;
-- 
2.25.1

