From ccb2c34d89c6724380990372b436c34547fc6de0 Mon Sep 17 00:00:00 2001
From: Mikolaj Staworzynski <mikolaj.staworzynski@redembedded.com>
Date: Tue, 24 Jan 2023 00:48:41 +0100
Subject: [OMWAPPI-1119] Cobalt vault introduced

---
 Source/cryptography/cryptography_vault_ids.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Source/cryptography/cryptography_vault_ids.h b/Source/cryptography/cryptography_vault_ids.h
index 0d3fb20..1eabf5d 100644
--- a/Source/cryptography/cryptography_vault_ids.h
+++ b/Source/cryptography/cryptography_vault_ids.h
@@ -28,5 +28,6 @@ extern "C"
         CRYPTOGRAPHY_VAULT_DEFAULT = 0,
         CRYPTOGRAPHY_VAULT_PLATFORM = 1,
         CRYPTOGRAPHY_VAULT_PROVISIONING = 0x10,
-        CRYPTOGRAPHY_VAULT_NETFLIX = 0x11
+        CRYPTOGRAPHY_VAULT_NETFLIX = 0x11,
+        CRYPTOGRAPHY_VAULT_COBALT = 0x12
     };
-- 
2.25.1

