#####################################################################
Date: Wed, 07 Sep 2022 10:35:00 PM
From:
Subject: Fix for Thunder trace log enable error  
Source: COMCAST
License:
Upstream-Status:
Signed-off-by: muhammed rafi c <muhammed_rafi@comcast.com>
#####################################################################
Index: git/Source/core/Netlink.cpp
===================================================================
--- git.orig/Source/core/Netlink.cpp
+++ git/Source/core/Netlink.cpp
@@ -22,6 +22,14 @@
 #include "Netlink.h"
 #include "Sync.h"
 
+#ifdef __WINDOWS__
+#include <Winsock2.h>
+#include <ws2tcpip.h>
+#define __ERRORRESULT__ ::WSAGetLastError()
+#else
+#define __ERRORRESULT__ errno
+#endif
+
 // #define DEBUG_FRAMES 1
 
 namespace WPEFramework {

