diff --git a/Source/bluetoothaudiosink/CMakeLists.txt b/Source/bluetoothaudiosink/CMakeLists.txt
index f6dbd96..7498eda 100644
--- a/Source/bluetoothaudiosink/CMakeLists.txt
+++ b/Source/bluetoothaudiosink/CMakeLists.txt
@@ -21,7 +21,7 @@ cmake_minimum_required(VERSION 3.0)
 find_package(WPEFramework)

 project(bluetoothaudiosink)
-project_version(1.0.0)
+project_version(4.1.0)

 set(TARGET ${PROJECT_NAME})

diff --git a/Source/compositorclient/CMakeLists.txt b/Source/compositorclient/CMakeLists.txt
index 1bc2b75..891aa20 100644
--- a/Source/compositorclient/CMakeLists.txt
+++ b/Source/compositorclient/CMakeLists.txt
@@ -21,7 +21,7 @@ cmake_minimum_required(VERSION 3.15)

 find_package(WPEFramework)

-project_version(1.0.0)
+project_version(4.1.0)

 set(TARGET ${PROJECT_NAME})

diff --git a/Source/cryptography/CMakeLists.txt b/Source/cryptography/CMakeLists.txt
index 38bb878..8b8e6b1 100644
--- a/Source/cryptography/CMakeLists.txt
+++ b/Source/cryptography/CMakeLists.txt
@@ -21,7 +21,7 @@ cmake_minimum_required(VERSION 3.3)

 find_package(WPEFramework)

-project_version(1.0.0)
+project_version(4.1.0)

 set(TARGET ${NAMESPACE}${PROJECT_NAME})

diff --git a/Source/deviceinfo/CMakeLists.txt b/Source/deviceinfo/CMakeLists.txt
index 7a966bb..1ed3278 100644
--- a/Source/deviceinfo/CMakeLists.txt
+++ b/Source/deviceinfo/CMakeLists.txt
@@ -21,7 +21,7 @@ cmake_minimum_required(VERSION 3.3)

 find_package(WPEFramework)

-project_version(1.0.0)
+project_version(4.1.0)

 set(TARGET ${PROJECT_NAME})

diff --git a/Source/displayinfo/CMakeLists.txt b/Source/displayinfo/CMakeLists.txt
index fb84cc7..5388500 100644
--- a/Source/displayinfo/CMakeLists.txt
+++ b/Source/displayinfo/CMakeLists.txt
@@ -21,7 +21,7 @@ cmake_minimum_required(VERSION 3.3)

 find_package(WPEFramework)

-project_version(1.0.0)
+project_version(4.1.0)

 set(TARGET ${PROJECT_NAME})

diff --git a/Source/localtracer/CMakeLists.txt b/Source/localtracer/CMakeLists.txt
index 818b07a..3392053 100644
--- a/Source/localtracer/CMakeLists.txt
+++ b/Source/localtracer/CMakeLists.txt
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 3.7)

 find_package(WPEFramework)

-project_version(1.0.0)
+project_version(4.1.0)

 set(MODULE_NAME ${NAMESPACE}LocalTracer)

@@ -45,4 +45,4 @@ HeaderOnlyInstallCMakeConfig(TARGET ${MODULE_NAME} TREAT_AS_NORMAL)

 if(EXAMPLE_LOCALTRACER)
     add_subdirectory(example)
-endif()
\ No newline at end of file
+endif()
diff --git a/Source/ocdm/CMakeLists.txt b/Source/ocdm/CMakeLists.txt
index b7526a0..d1e3de8 100644
--- a/Source/ocdm/CMakeLists.txt
+++ b/Source/ocdm/CMakeLists.txt
@@ -21,7 +21,7 @@ cmake_minimum_required(VERSION 3.3 )

 find_package(WPEFramework)

-project_version(1.0.0)
+project_version(4.1.0)

 set(TARGET ${PROJECT_NAME})

diff --git a/Source/playerinfo/CMakeLists.txt b/Source/playerinfo/CMakeLists.txt
index 2eda446..023dc5c 100644
--- a/Source/playerinfo/CMakeLists.txt
+++ b/Source/playerinfo/CMakeLists.txt
@@ -21,7 +21,7 @@ cmake_minimum_required(VERSION 3.3)

 find_package(WPEFramework)

-project_version(1.0.0)
+project_version(4.1.0)

 set(TARGET ${PROJECT_NAME})

diff --git a/Source/provisionproxy/CMakeLists.txt b/Source/provisionproxy/CMakeLists.txt
index 8e9fbd8..67854fc 100644
--- a/Source/provisionproxy/CMakeLists.txt
+++ b/Source/provisionproxy/CMakeLists.txt
@@ -21,7 +21,7 @@ cmake_minimum_required(VERSION 3.3)

 find_package(WPEFramework)

-project_version(1.0.0)
+project_version(4.1.0)

 set(TARGET ${PROJECT_NAME})

diff --git a/Source/securityagent/CMakeLists.txt b/Source/securityagent/CMakeLists.txt
index c9dea3a..a8f8bbd 100644
--- a/Source/securityagent/CMakeLists.txt
+++ b/Source/securityagent/CMakeLists.txt
@@ -21,7 +21,7 @@ cmake_minimum_required(VERSION 3.3)

 find_package(WPEFramework)

-project_version(1.0.0)
+project_version(4.1.0)

 set(TARGET ${PROJECT_NAME})

diff --git a/Source/virtualinput/CMakeLists.txt b/Source/virtualinput/CMakeLists.txt
index b6ab243..ae3149c 100644
--- a/Source/virtualinput/CMakeLists.txt
+++ b/Source/virtualinput/CMakeLists.txt
@@ -21,7 +21,7 @@ cmake_minimum_required(VERSION 3.3)

 find_package(WPEFramework)

-project_version(1.0.0)
+project_version(4.1.0)

 set(TARGET ${NAMESPACE}${PROJECT_NAME})



 
