From: Mikolaj Staworzynski <mikolaj.staworzynski@redembedded.com>
Date: Wed, 4 Jun 2025 12:12:37 +0200
Subject: [PATCH] ARRISEOS-47853: setup of group read permissions for logging
 config file

---
 Source/messaging/MessageUnit.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Source/messaging/MessageUnit.h b/Source/messaging/MessageUnit.h
index 389fe8b..d42ff8a 100644
--- a/Source/messaging/MessageUnit.h
+++ b/Source/messaging/MessageUnit.h
@@ -438,6 +438,9 @@ namespace WPEFramework {
 
                     Core::File file(settingsFilename);
                     file.Create();
+                    if (file.Permission(Core::File::USER_READ | Core::File::USER_WRITE | Core::File::GROUP_READ) != Core::ERROR_NONE) {
+                        TRACE_L1("Failed to update permissions for %s file\n", settingsFilename);
+                    }
                     auto written = file.Write(reinterpret_cast<const uint8_t*>(settings.data()), settings.size());
 
                     auto *group_info = getgrnam(wpeframeworkAccessGroupName);
-- 
2.43.0

