#####################################################################
Date: Fri, 09 Dec 2022 12:35:00 PM
From: Karunakaran A
Subject: Fix for Thunder trace log enable error  
Source: COMCAST
License: Apache-2.0
Upstream-Status: Completed
Signed-off-by: Karunakaran_Amirthalingam@cable.comcast.com
#####################################################################
diff --git a/Source/core/JSON.h b/Source/core/JSON.h
index 149ae392..32344875 100644
--- a/Source/core/JSON.h
+++ b/Source/core/JSON.h
@@ -1623,7 +1623,8 @@ namespace Core {
             inline const string Value() const
             {
                 if ((_flagsAndCounters & (SetBit | QuoteFoundBit | QuotedSerializeBit)) == (SetBit | QuoteFoundBit)) {
-                    return ('\"' + Core::ToString(_value.c_str()) + '\"');
+                    return (Core::ToQuotedString('\"', _value));
+
                 }
                 return (((_flagsAndCounters & (SetBit | NullBit)) == SetBit) ? Core::ToString(_value.c_str()) : Core::ToString(_default.c_str()));
             }
diff --git a/Source/core/Serialization.cpp b/Source/core/Serialization.cpp
index f3beb69b..399bd9ea 100644
--- a/Source/core/Serialization.cpp
+++ b/Source/core/Serialization.cpp
@@ -289,6 +289,22 @@ POP_WARNING()
         length = filler;

         return (index);
+
+    }
+
+    // If we are going to mark the length of a string by quotes, make
+    // sure that the internal quotes are escaped...
+    string EXTERNAL ToQuotedString(const TCHAR quote, const string& input) {
+        string result;
+        result += quote;
+        for (auto entry : input) {
+            if (entry == quote) {
+                result += '\\';
+            }
+            result += entry;
+        }
+        result += quote;
+        return (result);
     }

     bool CodePointToUTF16(const uint32_t codePoint, uint16_t& lowPart, uint16_t& highPart) {
diff --git a/Source/core/Serialization.h b/Source/core/Serialization.h
index 659c8cf5..eb55d563 100644
--- a/Source/core/Serialization.h
+++ b/Source/core/Serialization.h
@@ -264,6 +264,8 @@ POP_WARNING()
     void EXTERNAL ToString(const uint8_t object[], const uint16_t length, const bool padding, string& result);

     uint16_t EXTERNAL FromString(const string& newValue, uint8_t object[], uint16_t& length, const TCHAR* ignoreList = nullptr);
+    string EXTERNAL ToQuotedString(const TCHAR quote, const string& input);
+

     //------------------------------------------------------------------------
     // Codepoint: Operations to extract and convert code points.
