diff --git a/Source/core/Trace.h b/Source/core/Trace.h
index c235efc0..8dfca934 100644
--- a/Source/core/Trace.h
+++ b/Source/core/Trace.h
@@ -174,7 +174,13 @@ namespace WPEFramework {

 #define VERIFY(expr) ASSERT(expr)
 #else
-#define ASSERT(x)
+
+#define ASSERT(expr)                                                                                            \
+    do {                                                                                                        \
+        if (!(expr)) {                                                                                          \
+            ASSERT_LOGGER("===== $$ [%d]: ASSERT [%s:%d] (%s)\n", TRACE_PROCESS_ID, __FILE__, __LINE__, #expr); \
+        }                                                                                                       \
+    } while(0)

 #define VERIFY(expr)                                                                                                   \
     do {                                                                                                               \
@@ -185,7 +191,13 @@ namespace WPEFramework {
 
 
 
-#define ASSERT_VERBOSE(x, y, ...)
+#define ASSERT_VERBOSE(expr, format, ...)                                                                                                            \
+    do {                                                                                                                                             \
+        if (!(expr)) {                                                                                                                               \
+            ASSERT_LOGGER("===== $$ [%d]: ASSERT [%s:%d] (%s)\n         " #format "\n", TRACE_PROCESS_ID, __FILE__, __LINE__, #expr, ##__VA_ARGS__); \
+        }                                                                                                                                            \
+    } while(0)
+
 #endif
 
 #define LOG(LEVEL, MESSAGE)                                                         \
diff --git a/Source/com/Administrator.cpp b/Source/com/Administrator.cpp
index 6fa2cde5..7c1d7e92 100644
--- a/Source/com/Administrator.cpp
+++ b/Source/com/Administrator.cpp
@@ -300,8 +300,10 @@ namespace RPC {
                     }
                 } while ((loop->Decrement()) && (result == Core::ERROR_NONE));

+#ifdef __DEBUG__
                 ASSERT (loop->Flushed() == true);

+#endif
                 loop++;
             }
             _channelReferenceMap.erase(remotes);
