diff --git a/cmake/common/CompileSettings.cmake b/cmake/common/CompileSettings.cmake
index e2258644..92027663 100644
--- a/cmake/common/CompileSettings.cmake
+++ b/cmake/common/CompileSettings.cmake
@@ -71,6 +71,7 @@ if("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
     target_compile_definitions(CompileSettings INTERFACE _THUNDER_DEBUG)
     target_compile_definitions(CompileSettings INTERFACE _THUNDER_CALLSTACK_INFO)
     target_compile_options(CompileSettings INTERFACE -funwind-tables)
+    target_compile_options(CompileSettings INTERFACE -Os)
     set(CONFIG_DIR "Debug" CACHE STRING "Build config directory" FORCE)

 elseif("${CMAKE_BUILD_TYPE}" STREQUAL "DebugOptimized")
@@ -89,6 +90,7 @@ elseif("${CMAKE_BUILD_TYPE}" STREQUAL "Release")
     target_compile_definitions(CompileSettings INTERFACE _THUNDER_NDEBUG)
     target_compile_definitions(CompileSettings INTERFACE _THUNDER_CALLSTACK_INFO)
     target_compile_options(CompileSettings INTERFACE -funwind-tables)
+    target_compile_options(CompileSettings INTERFACE -Os)
     set(CONFIG_DIR "Release" CACHE STRING "Build config directory" FORCE)

 elseif("${CMAKE_BUILD_TYPE}" STREQUAL "MinSizeRel")
