diff --git a/Source/cryptography/implementation/SecApi/Cipher.cpp b/Source/cryptography/implementation/SecApi/Cipher.cpp
index 64c9800..4b485f4 100644
--- a/Source/cryptography/implementation/SecApi/Cipher.cpp
+++ b/Source/cryptography/implementation/SecApi/Cipher.cpp
@@ -41,7 +41,7 @@ namespace Implementation {
         ASSERT(vault != nullptr);
         ASSERT(keyId != 0);
         ASSERT(keyLength != 0);
-        ASSERT(ivLength != 0)
+        ASSERT(ivLength != 0);

     }
