From 780b42e937b406a866595071fb0e6b01cb79605d Mon Sep 17 00:00:00 2001
From: Dawid Trendota <dawid.trendota@consult.red>
Date: Wed, 17 May 2023 16:53:32 +0200
Subject: [PATCH] 0116-OMWAPPI-1634-Implementation-of-IMediaSystemMetrics

---
 MediaSession.cpp | 14 ++++++++++++++
 MediaSession.h   |  6 +++++-
 MediaSystem.cpp  | 21 ++++++++++++++++++++-
 3 files changed, 39 insertions(+), 2 deletions(-)

Index: OCDM-Widevine-Nexus-SVP/MediaSession.cpp
===================================================================
--- OCDM-Widevine-Nexus-SVP.orig/MediaSession.cpp
+++ OCDM-Widevine-Nexus-SVP/MediaSession.cpp
@@ -655,4 +655,22 @@ CDMi_RESULT MediaKeySession::ReleaseClea
     uint8_t  *f_pbClearContentOpaque ){
   return CDMi_SUCCESS;
 }
+
+CDMi_RESULT MediaKeySession::Metrics(uint32_t& bufferLength, uint8_t buffer[]) const {
+#if CENC_VERSION >= 16
+    std::string metrics;
+    if(widevine::Cdm::kSuccess == m_cdm->getMetrics(&metrics)) {
+        if(bufferLength < metrics.size()) {
+            return CDMi_BUFFER_TOO_SMALL;
+        }
+        bufferLength = metrics.size();
+        memcpy(buffer, reinterpret_cast<const uint8_t*>(metrics.c_str()), bufferLength);
+        return CDMi_SUCCESS;
+    }
+#else
+    bufferLength = 0;
+#endif
+    return CDMi_S_FALSE;
+}
+
 }  // namespace CDMi
Index: OCDM-Widevine-Nexus-SVP/MediaSession.h
===================================================================
--- OCDM-Widevine-Nexus-SVP.orig/MediaSession.h
+++ OCDM-Widevine-Nexus-SVP/MediaSession.h
@@ -69,7 +69,7 @@
 
 namespace CDMi
 {
-class MediaKeySession : public IMediaKeySession
+class MediaKeySession : public IMediaKeySession, public IMediaSessionMetrics
 {
 public:
     MediaKeySession(widevine::Cdm*, int32_t);
@@ -128,6 +128,10 @@ public:
     void onDirectIndividualizationRequest(const std::string&, const std::string&);
 #endif
 
+    // IMediaSessionMetrics overrides
+    // ------------------------------------------------------------------------------------------
+    CDMi_RESULT Metrics (uint32_t& bufferLength, uint8_t buffer[]) const override;
+
 private:
     void onKeyStatusError(widevine::Cdm::Status status);
 
Index: OCDM-Widevine-Nexus-SVP/MediaSystem.cpp
===================================================================
--- OCDM-Widevine-Nexus-SVP.orig/MediaSystem.cpp
+++ OCDM-Widevine-Nexus-SVP/MediaSystem.cpp
@@ -59,7 +59,7 @@ using namespace WPEFramework;
 
 namespace CDMi {
 
-class WideVine : public IMediaKeys, public widevine::Cdm::IEventListener
+class WideVine : public IMediaKeys, public IMediaSystemMetrics, public widevine::Cdm::IEventListener
 {
 private:
     WideVine (const WideVine&) = delete;
@@ -358,6 +358,29 @@ public:
         _adminLock.Unlock();
     }
 #endif
+
+    // IMediaSystemMetrics overrides
+    // ------------------------------------------------------------------------------------------
+    CDMi_RESULT Metrics (uint32_t& bufferLength, uint8_t buffer[]) const override {
+#if CENC_VERSION >= 16
+        std::string metrics;
+        if(_cdm == nullptr) {
+            return CDMi_S_FALSE;
+        }
+        if(widevine::Cdm::kSuccess == _cdm->getMetrics(&metrics)) {
+            if(bufferLength < metrics.size()) {
+                return CDMi_BUFFER_TOO_SMALL;
+            }
+            bufferLength = metrics.size();
+            memcpy(buffer, reinterpret_cast<const uint8_t*>(metrics.c_str()), bufferLength);
+            return CDMi_SUCCESS;
+        }
+#else
+	bufferLength = 0;
+#endif
+        return CDMi_S_FALSE;
+    }
+
 private:
     WPEFramework::Core::CriticalSection _adminLock;
     widevine::Cdm* _cdm;
