diff --git a/CMakeLists.txt b/CMakeLists.txt
index c88235f..9fd5211 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,7 +20,7 @@ cmake_minimum_required(VERSION 3.12)
 project(Interfaces)
 find_package(WPEFramework)
 
-project_version(1.0.0)
+project_version(4.1.0)
 
 set(INTERFACES_PATTERNS "I*.h" CACHE STRING "Patterns matching files for which stubs should be generated")
 set(JSONRPC_PATTERNS "*.json" CACHE STRING "Patterns matching files for which json stubs should be generated")
-- 
2.25.1

