diff --git a/Source/cryptography/tests/cryptography_test/CMakeLists.txt b/Source/cryptography/tests/cryptography_test/CMakeLists.txt
index 5e54760..2196e59 100644
--- a/Source/cryptography/tests/cryptography_test/CMakeLists.txt
+++ b/Source/cryptography/tests/cryptography_test/CMakeLists.txt
@@ -131,6 +131,7 @@ target_compile_options(cobalttests PUBLIC -DVALIDATE_VALUES -DTEST_PLATFORM_APOL
 target_link_libraries(cobalttests
         PRIVATE
         ${NAMESPACE}Cryptography
+        ${NAMESPACE}Core::${NAMESPACE}Core
         ssl
         crypto
         utility
@@ -157,6 +158,7 @@ target_compile_options(lgiplatformtest PUBLIC -DVALIDATE_VALUES -DTEST_PLATFORM_
 target_link_libraries(lgiplatformtest
         PRIVATE
         ${NAMESPACE}Cryptography
+        ${NAMESPACE}Core::${NAMESPACE}Core
         ssl
         crypto
         utility
