From 86622a025435aaaf7df5cc39c3310018b8cb4b89 Mon Sep 17 00:00:00 2001
From: Ievgen Mutavchi <Ievgen_Mutavchi@comcast.com>
Date: Wed, 7 Apr 2021 14:08:12 -0400
Subject: [PATCH] Reduce BrowserConsoleLog

---
 WebKitBrowser/WebKitImplementation.cpp | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/WebKitBrowser/WebKitImplementation.cpp b/WebKitBrowser/WebKitImplementation.cpp
index 68769d0..6f6151f 100644
--- a/WebKitBrowser/WebKitImplementation.cpp
+++ b/WebKitBrowser/WebKitImplementation.cpp
@@ -70,6 +70,8 @@ WK_EXPORT WKProcessID WKPageGetProcessIdentifier(WKPageRef page);
 namespace WPEFramework {
 namespace Plugin {
 
+    static string consoleLogPrefix;
+
 #ifndef WEBKIT_GLIB_API
     static void onDidReceiveSynchronousMessageFromInjectedBundle(WKContextRef context, WKStringRef messageName,
         WKTypeRef messageBodyObj, WKTypeRef* returnData, const void* clientInfo);
@@ -227,8 +229,12 @@ namespace Plugin {
         nullptr, // runBeforeUnloadConfirmPanel
         nullptr, // fullscreenMayReturnToInline
         // willAddDetailedMessageToConsole
-        [](WKPageRef, WKStringRef source, WKStringRef, uint64_t line, uint64_t column, WKStringRef message, WKStringRef, const void* clientInfo) {
-            TRACE_GLOBAL(BrowserConsoleLog, (message, line, column));
+        [](WKPageRef, WKStringRef, WKStringRef, uint64_t line, uint64_t column, WKStringRef message, WKStringRef url, const void* clientInfo) {
+          if (WPEFramework::Trace::TraceType<BrowserConsoleLog, &WPEFramework::Core::System::MODULE_NAME>::IsEnabled() == false)
+            return;
+          string urlStr = WebKit::Utils::WKStringToString(url);
+          string messageStr = WebKit::Utils::WKStringToString(message);
+          fprintf(stderr, "[%s]:%s:%llu,%llu %s\n", consoleLogPrefix.c_str(), Core::FileNameOnly(urlStr.c_str()), line, column, messageStr.c_str());
         },
     };
 
@@ -1439,6 +1445,7 @@ static GSourceFuncs _handlerIntervention =
 
         uint32_t Configure(PluginHost::IShell* service) override
         {
+            consoleLogPrefix = service->Callsign();
             _service = service;
 
             _dataPath = service->DataPath();
-- 
2.7.4

