diff --git a/Source/core/CallsignTLS.cpp b/Source/core/CallsignTLS.cpp
index 870896bf..3172697d 100644
--- a/Source/core/CallsignTLS.cpp
+++ b/Source/core/CallsignTLS.cpp
@@ -20,7 +20,7 @@
 #include "CallsignTLS.h"
 #include "Thread.h"
 
-#ifdef __CORE_WARNING_REPORTING__
+//#ifdef __CORE_WARNING_REPORTING__
 
 namespace WPEFramework {
 namespace Core {
@@ -41,4 +41,4 @@ namespace Core {
 }
 } 
 
-#endif
+//#endif

